syntax = "proto3";

import "google/protobuf/empty.proto";

service PortfolioUpdates {
    rpc core_alive(google.protobuf.Empty) returns (google.protobuf.Empty) {}
    rpc enqueue_contracts(stream GrpcContract) returns (google.protobuf.Empty) {}
}

message GrpcContract {
    string symbol = 1;
    string expiry = 2;
    double strike = 3;
    string type = 4;
}